handle documented cases in Schema annotation

NecroKote opened the pull request:
- `minLength` and `maxLength` are ignored if the value is below 0
- `format` is included only if it's not blank

Closes #29

Commit:
handle documented cases in Schema annotation

- `minLength` and `maxLength` are ignored if the value is below 0
- `minLength` and `maxLength` are not included for anything other than `string` type
- `format` is included only if it's not blank

rmcmk reviewed (changes requested):
Hey -- nice fix. Just wanted to point out there are a two more `@Schema` fields that kenerator supports, with non-respected ignore conditions. Apologies, I probably should have stated all affected schema attributes in #29  

```java
    /**
     * Sets the maximum numeric value for a property.  Ignored if the value is an empty string.
     *
     * @return the maximum value for this schema
     **/
    String maximum() default "";
    
    /**
     * Sets the minimum numeric value for a property.  Ignored if the value is an empty string or not a number.
     *
     * @return the minimum value for this schema
     **/
    String minimum() default "";
```

On a side note, `ignored if the value is an empty string or not a number.` is only stated on `minimum` but is likely the case for both.

NecroKote commented:
re: `minimum` and `maximum` - those are already covered by existing `.filter { it.isNotBlank() }`
At least for the "Ignored if the value is an empty string" part

Commit:
revert: minLength and maxLength are not included for anything other than string type

original commit contained an error, but also for more sophisticated cases it's not clear how to determine resulting type

rmcmk commented:
:facepalm: Looked at this code half a dozen times and missed that LOL. Guess that only leaves the number validation bit. Not sure if that is necessary, though.

SMILEY4 commented:
Looks good to me  :+1:    Thank you!
If its ready i'll merge it.

NecroKote commented:
Please do!
